cmd/dockerd/daemon.go

```diff
@@ -380,12 +380,20 @@ func (cli *DaemonCli) start(opts *daemonOptions) (err error) {
 // TODO: This can be removed after buildkit is updated to use http/protobuf as the default.
 func setOTLPProtoDefault() {
 	const (
-		tracesEnv = "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL"
-		protoEnv  = "OTEL_EXPORTER_OTLP_PROTOCOL"
+		tracesEnv  = "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL"
+		metricsEnv = "OTEL_EXPORTER_OTLP_METRICS_PROTOCOL"
+		protoEnv   = "OTEL_EXPORTER_OTLP_PROTOCOL"
+
+		defaultProto = "http/protobuf"
 	)
 
-	if os.Getenv(tracesEnv) == "" && os.Getenv(protoEnv) == "" {
-		os.Setenv(tracesEnv, "http/protobuf")
+	if os.Getenv(protoEnv) == "" {
+		if os.Getenv(tracesEnv) == "" {
+			os.Setenv(tracesEnv, defaultProto)
+		}
+		if os.Getenv(metricsEnv) == "" {
+			os.Setenv(metricsEnv, defaultProto)
+		}
```

Comment:
For context: we need this because the OTEL default now for `OTEL_EXPORTER_OTLP_PROTOCOL` is `http/protobuf` (see https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md and https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specify-protocol).

while buildkit defaults to `grpc` due to backwards compatibility (which is also compliant with the spec):

> If no configuration is provided the default transport SHOULD be `http/protobuf`
unless SDKs have good reasons to choose `grpc` as the default (e.g. for backward
compatibility reasons when `grpc` was already the default in a stable SDK
release).

Reply:
> This seems fine to me, but in the future I'd like to work on decoupling moby's reliance on buildkit for its opentelemetry configuration.

:100: I think we started using the BuildKit package to make sure we were aligned on how things are set up (and configured), but now that we have more and more projects integrating OTEL, we should look where things go. It all is "vanilla" / "standard" OTEL bits, then we may not need anything central, but if there's certain conventions / approaches we want to make sure are kept "in sync", then _maybe_ we need some utility module with those shared parts.

Happy to hear thoughts / get input on that though (I'm really bad at all the OTEL stuff)